algo/src/main/kotlin/com/hartwig/actin/algo/TrialMatcher.kt

```diff
@@ -44,10 +46,24 @@ class TrialMatcher(private val evaluationFunctionFactory: EvaluationFunctionFact
 
     private fun evaluateEligibility(patient: PatientRecord, eligibility: List<Eligibility>): Map<Eligibility, Evaluation> {
         return eligibility.sortedWith(EligibilityComparator()).associateWith {
-            evaluationFunctionFactory.create(it.function).evaluate(patient)
+            combineMessages(evaluationFunctionFactory.create(it.function).evaluate(patient))
```

Comment:
I think an extension function on `Evaluation` would be nice here so you could do:
```suggestion
           evaluationFunctionFactory.create(it.function).evaluate(patient).combineMessages()
```